std/guards/README.md

```diff
@@ -0,0 +1,307 @@
+# guards
+
+A comprehensive collection of type guards.
+
+## Table of contents
+
+-   [Usage](#usage)
+
+    -   [Primitives](#primitives)
+
+        -   [`isBigInt`](#isbigint)
+        -   [`isBoolean`](#isboolean)
+        -   [`isNumber`](#isnumber)
+        -   [`isString`](#isstring)
+        -   [`isSymbol`](#issymbol)
+        -   [`isUndefined`](#isundefined)
+
+    -   [Special](#special)
+
+        -   [`isNull`](#isnull)
+        -   [`isFunction`](#isfunction)
+        -   [`isObject`](#isobject)
+        -   [`isArray`](#isarray)
+        -   [`isMap`](#ismap)
+        -   [`isSet`](#isset)
+        -   [`isWeakMap`](#isweakmap)
+        -   [`isWeakSet`](#isweakset)
+        -   [`isDate`](#isdate)
+
+    -   [Convenience](#convenience)
+
+        -   [`isNonEmptyArray`](#isnonemptyarray)
+        -   [`isValidNumber`](#isvalidnumber)
+        -   [`isInteger`](#isinteger)
+        -   [`isPositiveInteger`](#ispositiveinteger)
+        -   [`isNonNegativeInteger`](#isnonnegativeinteger)
+        -   [`isNegativeInteger`](#isnegativeinteger)
+
+## Usage
+
+From <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures>:
+
+The latest ECMAScript standard defines nine types:
+
+-   Six Data Types that are primitives, checked by `typeof` operator:
+    -   `undefined`: `typeof instance === "undefined"`
+    -   `Boolean`: `typeof instance === "boolean"`
+    -   `Number`: `typeof instance === "number"`
+    -   `String`: `typeof instance === "string"`
+    -   `BigInt`: `typeof instance === "bigint"`
+    -   `Symbol`: `typeof instance === "symbol"`
+-   `null`: `typeof instance === "object"`. Special primitive type having
+    additional usage for it's value: if object is not inherited, then `null` is
+    shown;
```

Comment:
This phrasing is very confusing and sounds like `null` isn't a primitive.

Reply:
This is a copy-paste from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures 🤔

Reply:
I can see that now. That page is very broken, I suggest not referring to it at all.

Reply:
Happy to remove it, just thought it would be useful to reference _something_ that describes JavaScript types comprehensively and most people can refer to.

That said, it might make more sense to reference https://www.typescriptlang.org/docs/handbook/basic-types.html.